### DISCUSSION TITLE
WSL 2 consumes massive amounts of RAM and temporary Solution

### DISCUSSION THREAD

**Discussion by wlya** [microsoft/WSL#6500] [General]
WSL 2 starts using huge amounts of RAM after a while, just using it like normal. Vmmem is using 7 GB of my 16 GB of RAM and not returning any, even though Ubuntu is actually using much less. 

```
sudo su
echo 3 > /proc/sys/vm/drop_caches
echo 2 > /proc/sys/vm/drop_caches
echo 1 > /proc/sys/vm/drop_caches
sync
```

**Comment by digitaldias** [comment]
This workaround is not working for me, on ubuntu 20.04 - is there a particular dialect of that I need to use for the workaround? Facing the same issues with memory btw.

**Comment by BoZenKhaa** [comment]
Could it be that the memory in WSL is used as cache, as in this [issue](https://github.com/microsoft/WSL/issues/4166)? From Windows, that memory will appear as used. 

If that's the case, you can [try this](https://www.linuxatemyram.com/). That worked for me in WSL2 with Ubuntu 18.04.

**Comment by wlya** [comment]
> This workaround is not working for me, on ubuntu 20.04 - is there a particular dialect of that I need to use for the workaround? Facing the same issues with memory btw.

I have tested Ubuntu20.04 wsl2, works fine.

**Comment by wlya** [comment]
> Could it be that the memory in WSL is used as cache, as in this [issue](https://github.com/microsoft/WSL/issues/4166)? From Windows, that memory will appear as used.

Yes, just because Linux does not want to free memory and intended to cache them as possible.

**Comment by craigloewen-msft** [comment]
I'm going to label this as a discussion. The best workaround here is to [use .wslconfig to put a memory limit on your WSL 2 VM](https://docs.microsoft.com/en-us/windows/wsl/wsl-config#configure-global-options-with-wslconfig). Please also see this longstanding thread where we are tracking this issue: https://github.com/microsoft/WSL/issues/4166

**Comment by wlya** [comment]
> I'm going to label this as a discussion. The best workaround here is to [use .wslconfig to put a memory limit on your WSL 2 VM](https://docs.microsoft.com/en-us/windows/wsl/wsl-config#configure-global-options-with-wslconfig). Please also see this longstanding thread where we are tracking this issue: #4166

Ok, thank you. I post this just to post another viewpoint.
